datajunction-clients/python/datajunction/nodes.py

```diff
@@ -420,13 +421,35 @@ def validate(self) -> str:
         return response["status"]
 
 
+@dataclass
+class NodeName(SerializableMixin):
+    """Node name"""
+
+    name: str
+
+
+@dataclass
+class DimensionLink(SerializableMixin):
+    """
+    Dimension join links
+    """
+
+    dimension: NodeName
```

Comment:
Why not simply a `dimension: str` ?

Reply:
Unfortunately the format of ORM-based dataclass serialization means that wouldn't work out of the box.